deploy/crds/push_v1alpha1_unifiedpushserver_crd.yaml

```diff
@@ -85,10 +85,31 @@ spec:
           type: object
         status:
           properties:
+            message:
+              description: Message is a more human-readable message indicating details
+                about current phase or error.
+              type: string
             phase:
+              description: Phase indicates whether the CR is reconciling(good), failing(bad),
+                or initializing.
               type: string
+            ready:
+              description: Ready is True if all resources are in a ready state and
+                all work is done (phase should be "reconciling").
+              type: boolean
+            secondaryResources:
+              additionalProperties:
+                items:
+                  type: string
+                type: array
+              description: 'SecondaryResources is a map of all the secondary resources
+                types and names created for this CR.  e.g "Deployment": [ "DeploymentName1",
+                "DeploymentName2" ]'
+              type: object
           required:
           - phase
+          - message
+          - ready
```

Comment:
@davidffrench I'm thinking that maybe these shouldn't be "required" for this version of the CRD, wdyt? The case I'm thinking of is where someone applies the CRDs but can't actually upgrade the operator immediately for one reason or another (or have to roll back) -- the existing version of the operator won't be able to reconcile the updated CRD if we make these required.

Reply:
I would love to know what happens in this scenario for 'required' status fields. The scenario you are outlining makes sense to not make them mandatory.

Reply:
I tested it out, and got the following error in one case:

<details>
{"level":"error","ts":1574341351.4474332,"logger":"kubebuilder.controller","msg":"Reconciler error","controller":"unifiedpushserver-controller","request":"mobile-unifiedpush/unifiedpush","error":"UnifiedPushServer.push.aerogear.org \"unifiedpush\" is invalid: []: Invalid value: map[string]interface {}{\"kind\":\"UnifiedPushServer\", \"metadata\":map[string]interface {}{\"clusterName\":\"\", \"annotations\":map[string]interface {}{\"kubectl.kubernetes.io/last-applied-configuration\":\"{\\\"apiVersion\\\":\\\"push.aerogear.org/v1alpha1\\\",\\\"kind\\\":\\\"UnifiedPushServer\\\",\\\"metadata\\\":{\\\"annotations\\\":{},\\\"generation\\\":1,\\\"name\\\":\\\"unifiedpush\\\",\\\"namespace\\\":\\\"mobile-unifiedpush\\\",\\\"resourceVersion\\\":\\\"\\\",\\\"selfLink\\\":\\\"/apis/push.aerogear.org/v1alpha1/namespaces/mobile-unifiedpush/unifiedpushservers/unifiedpush\\\",\\\"uid\\\":\\\"\\\"},\\\"spec\\\":{\\\"useMessageBroker\\\":false}}\\n\"}, \"resourceVersion\":\"245971\", \"name\":\"unifiedpush\", \"namespace\":\"mobile-unifiedpush\", \"uid\":\"2306dd4a-0c5f-11ea-9c2f-16c554726cf3\", \"generation\":1, \"creationTimestamp\":\"2019-11-21T13:02:13Z\"}, \"status\":map[string]interface {}{\"phase\":\"Provisioning\"}, \"spec\":map[string]interface {}{\"useMessageBroker\":false}, \"apiVersion\":\"push.aerogear.org/v1alpha1\"}: validation failure list:\nstatus.message in body is required\nstatus.ready in body is required","stacktrace":"github.com/aerogear/unifiedpush-operator/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\t/home/jenkins/workspace/unifiedpush-operator_0.2.0/src/github.com/aerogear/unifiedpush-operator/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/aerogear/unifiedpush-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/jenkins/workspace/unifiedpush-operator_0.2.0/src/github.com/aerogear/unifiedpush-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:217\ngithub.com/aerogear/unifiedpush-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1\n\t/home/jenkins/workspace/unifiedpush-operator_0.2.0/src/github.com/aerogear/unifiedpush-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158\ngithub.com/aerogear/unifiedpush-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/home/jenkins/workspace/unifiedpush-operator_0.2.0/src/github.com/aerogear/unifiedpush-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\ngithub.com/aerogear/unifiedpush-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/home/jenkins/workspace/unifiedpush-operator_0.2.0/src/github.com/aerogear/unifiedpush-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134\ngithub.com/aerogear/unifiedpush-operator/vendor/k8s.io/apimachinery/pkg/util/wait.Until\n\t/home/jenkins/workspace/unifiedpush-operator_0.2.0/src/github.com/aerogear/unifiedpush-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88"}
</details>

^ which basically boils down to `validation failure list:\nstatus.message in body is required\nstatus.ready in body is required` if you care to look closely enough.

I'll update these to make them optional :)